awsdms: fix identifiers to be tagged by build version

otan opened the pull request:
When the branch got cut, AWS DMS tests were running in parallel, which
with static identifiers causes tests to collide with each other and fail
as they try deleting each other's instances.

To fix this, we now use the build version as part of the identifiers
which hopefully makes this unique per branch.

Release justification: test only change

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87825)
<!-- Reviewable:end -->

otan committed:
awsdms: fix identifiers to be tagged by build version

When the branch got cut, AWS DMS tests were running in parallel, which
with static identifiers causes tests to collide with each other and fail
as they try deleting each other's instances.

To fix this, we now use the build version as part of the identifiers
which hopefully makes this unique per branch.

Release justification: test only change

Release note: None

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/87825#-:-NBmE_ArEM0sJbHthopw:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
